src/Model/PodAwareStatusProvider.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Kubernetes\Client\Model;
+
+use Kubernetes\Client\Exception\PodNotFound;
+
+/**
+ * Provide up to date status information of a pod.
+ *
+ * @package Kubernetes\Client\Repository
+ */
+interface PodAwareStatusProvider
```

Comment:
I guess you should remove the `setPod` (or move the `$pod` to the method's arguments) and rename this interface `PodStatusProvider`